geomstats/visualization.py

```diff
@@ -490,6 +494,186 @@ def draw(self, ax, **kwargs):
+        self.ub = gs.array([[.5, gs.sqrt(3) / 2],
+                            [.5, -gs.sqrt(3) / 2],
+                            [-1., 0.]]) / gs.sqrt(3)
+
+        # (ua,na) is a positively oriented othonormal basis of
+        # the horizontal space at north pole
+        self.na = self.ub - S32.ambient_metric.inner_product(
+            self.ub, self.ua) * self.ua
+        self.na = self.na / S32.ambient_metric.norm(self.na)
+
+        if points is not None:
+            self.add_points(points)
+
+    def set_ax(self, ax=None):
+        """Set axis."""
+        if ax is None:
+            ax = plt.subplot(111, projection='3d')
+
+        ax_s = .5
+        plt.setp(ax,
+                 xlim=(-ax_s, ax_s),
+                 ylim=(-ax_s, ax_s),
+                 zlim=(-ax_s, ax_s),
+                 xlabel='X', ylabel='Y', zlabel='Z')
+        self.ax = ax
+
+    def set_view(self, elev=30., azim=90.):
+        """Set azimuth and elevation angle."""
+        self.elev, self.azim = gs.pi * elev / 180, gs.pi * azim / 180
+        self.ax.view_init(elev, azim)
+
+    def convert_to_polar_coordinates(self, points):
+        """Assign polar coordinates to given pre-shapes."""
+        aligned_points = S32.align(points, self.can)
+        speeds = S32.ambient_metric.log(aligned_points, self.can)
+
+        coords_theta = gs.arctan2(
+            S32.ambient_metric.inner_product(speeds, self.na),
+            S32.ambient_metric.inner_product(speeds, self.ua))
+        coords_phi = 2 * S32.ambient_metric.dist(self.can, aligned_points)
+
+        return coords_theta, coords_phi
+
+    def convert_to_spherical_coordinates(self, points):
+        """Convert polar coordinates to spherical one."""
+        coords_theta, coords_phi = self.convert_to_polar_coordinates(points)
+        coords_x = .5 * gs.cos(coords_theta) * gs.sin(coords_phi)
+        coords_y = .5 * gs.sin(coords_theta) * gs.sin(coords_phi)
+        coords_z = .5 * gs.cos(coords_phi)
+        spherical_coords = gs.transpose(gs.vstack(
+            [coords_x, coords_y, coords_z]))
+        return spherical_coords
+
+    def add_points(self, points):
+        """Add points to draw on the Kendall sphere."""
+        if self.point_type == 'extrinsic':
+            if not gs.all(M32.belongs(points)):
+                raise ValueError(
+                    'Points do not belong to Matrices(3, 2).')
+            points = S32.projection(points)
+        elif self.point_type == 'pre-shape' \
+                and not gs.all(S32.belongs(points)):
+            raise ValueError('Points do not belong to the pre-shape space.')
+        points = self.convert_to_spherical_coordinates(points)
+        if not isinstance(points, list):
+            points = list(points)
+        self.points.extend(points)
+
+    def clear_points(self):
+        """Clear the points to draw."""
+        self.points = []
+
+    def draw(self, n_theta=25, n_phi=13, scale=.05):
```

Comment:
Would it make sense to put:
```
def draw(self, n_theta=25, n_phi=13, scale=.05, elev=60., azim=0.):
    self.set_ax()
    self.set_view(elev=elev, azim=azim)
```
here? It would avoid repeating the lines:
```
S.set_ax()
S.set_view(60., 0.)
S.draw()
```
in the notebook.